src/controllers/store-exposure-scan-data.js

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { activateProfile, createProfile, parseExposureScanData } from '../external/onerep.js'
+import { setOnerepProfileId } from '../db/tables/subscribers.js'
+
+/**
+ * @typedef {{ success: false }} StoreExposureScanDataErrorResponse
+ * @typedef {{ success: true }} StoreExposureScanDataSuccessResponse
+ * @typedef {StoreExposureScanDataErrorResponse | StoreExposureScanDataSuccessResponse} StoreExposureScanDataResponse
+ */
+
+/** @type {import('express').RequestHandler<import('../external/onerep').ProfileData, StoreExposureScanDataResponse>} */
+async function storeExposureScanData (req, res, next) {
+  if (req.method !== 'POST' || !req.user) {
+    return next()
+  }
+  if (typeof req.user.onerep_profile_id === 'number') {
+    // Exposure scan data is already set for this user
+    res.status(409).send({ success: false })
+    return
+  }
+  const exposureScanInput = parseExposureScanData(req.body)
+  if (exposureScanInput === null) {
+    res.status(400).send({ success: false })
+    return
+  }
+
+  try {
+    const onerepProfileId = await createProfile(exposureScanInput)
+    await activateProfile(onerepProfileId)
```

Comment:
When a user enters their data but is not a Premium user yet, we're supposed to do a single scan, without monitoring yet. The API docs mention that the first profile activation triggers a scan (with `initial` as the `reason`), but it's not clear to me if that also activates monitoring and automatic opt-out. If so, we probably don't want to do this here, but I'm not sure if we can then trigger a one-time scan.

Reply:
Updated, because I found the [Scan Pathway](https://docs.onerep.com/#section/Scan-Pathway) docs, which provided the answer (you can scan without activating the profile).